src/core/execute.c

```diff
@@ -2385,7 +2385,10 @@ static int exec_child(
 
         exec_context_tty_reset(context, params);
 
-        if (!manager_is_confirm_spawn_disabled(unit->manager)) {
+        /* For some reasons units remaining in the same process group
+         * as PID 1 fail to acquire the console even if it's not used
+         * by any process. So skip the confirmation question for them. */
+        if (!context->same_pgrp && !manager_is_confirm_spawn_disabled(unit->manager)) {
```

Comment:
i think it might be nicer to change `manager_is_confirm_spawn_disabled(Manage *m)` into a `unit_shall_confirm_spawn(Unit *u)` and then hide the same_pgrp check in it?